thunder/tests/test_networks.py

```diff
@@ -558,6 +558,189 @@ def test_hf_llama():
+            57.95000457763672,
+            59.29000473022461,
+            59.77000427246094,
+            59.920005798339844,
+            61.190006256103516,
+            61.96000671386719,
+            62.50000762939453,
+            63.3700065612793,
+            63.48000717163086,
+            63.48000717163086,
+            63.66000747680664,
+            63.850006103515625,
+            64.08000946044922,
+            64.760009765625,
+            64.80001068115234,
+            64.81001281738281,
+            64.81001281738281,
+        ],
+        "short_factor": [
+            1.05,
+            1.05,
+            1.05,
+            1.1,
+            1.1,
+            1.1,
+            1.2500000000000002,
+            1.2500000000000002,
+            1.4000000000000004,
+            1.4500000000000004,
+            1.5500000000000005,
+            1.8500000000000008,
+            1.9000000000000008,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.000000000000001,
+            2.1000000000000005,
+            2.1000000000000005,
+            2.2,
+            2.3499999999999996,
+            2.3499999999999996,
+            2.3499999999999996,
+            2.3499999999999996,
+            2.3999999999999995,
+            2.3999999999999995,
+            2.6499999999999986,
+            2.6999999999999984,
+            2.8999999999999977,
+            2.9499999999999975,
+            3.049999999999997,
+            3.049999999999997,
+            3.049999999999997,
+        ],
+        "type": "su",
+    },
+    "rope_theta": 10000.0,
+    "sliding_window": 131072,
+    "tie_word_embeddings": False,
+    "torch_dtype": "bfloat16",
+    "transformers_version": "4.38.1",
+    "use_cache": True,
+    "vocab_size": 32064,
+    "_attn_implementation": "sdpa",
+}
+
+
+@requiresCUDA
+def test_hf_phi3_vision():
+    # This test takes around 4045406208 bytes (~4GB) of memory.
+    # Shapes for data generated with help of the following script
+    # https://github.com/microsoft/PhiCookBook/blob/main/code/03.Finetuning/Phi-3-vision-Trainingscript.py
+    from transformers import AutoModelForCausalLM
+    from transformers.models.phi3 import Phi3Config
+    from thunder.dynamo import thunderfx
+
+    cfg = Phi3Config(**phi3_vision_cfg)
+    cfg.num_hidden_layers = 2
+
+    with torch.device("cuda"):
+        model = AutoModelForCausalLM.from_config(cfg, trust_remote_code=False, torch_dtype=torch.bfloat16)
```

Comment:
Changing `vocab_size` from 32064 to a smaller number should decrease the memory requirements of this test.